addons/mail/static/src/models/thread/thread.js

```diff
@@ -1348,6 +1393,15 @@ function factory(dependencies) {
          * interface and to notify the server of the new state.
          */
         pendingFoldState: attr(),
+        /**
+         * Determine if there is a pending pin state change, which is a change
+         * of pin state requested by the client but not yet confirmed by the
+         * server.
+         *
+         * This field can be updated to immediately change the pin state on the
+         * interface and to notify the server of the new state.
+         */
+        pendingPinState: attr(),
```

Comment:
Now that these 2 are booleans, I think we should review their name.

I don't think the "state" part is necessary (it was coming from fold_state which is the actual field name).
And it should start with "is" or "has".

@alexkuhn any idea?

isPinnedPending?
isPendingPinned?
isPinned(On)Server?
isServerPinned?

Reply:
`serverPinState` => `isServerPinned`
As for `pendingPinState`, it's not really a pure boolean ATM, because there's the `undefined` value to mean "don't consider pending pinned". Would introduce 3rd field to determine whether there's a pending pin or not, because enforcing typing on `attr` is good:

`pendingPinState` => `isPendingPinned`
`hasPendingPinned` to determine if there's an active pending `isPinned`.

This may look confusing to have a similar field just different by `has` and `is`, and both being boolean... For me the nuance of "is" and "has" makes sense, that's why most booleans in messaging code have well-thought choice of verb as prefix... but if that's hard for other developers, this is maybe not a good decision...

Reply:
I like using `undefined` better in this case, or if we really want a hard value, `null`.

I feel like adding another variable just makes everything more complex. Not to mention the validity of whatever is inside `isPendingPinned` becomes linked to `hasPendingPinned` being true, and when `hasPendingPinned` is false `isPendingPinned` becomes very dangerous to read, which will not be obvious at all.

Reply:
> I like using `undefined` better in this case, or if we really want a hard value, `null`.
> 
> I feel like adding another variable just makes everything more complex. Not to mention the validity of whatever is inside `isPendingPinned` becomes linked to `hasPendingPinned` being true, and when `hasPendingPinned` is false `isPendingPinned` becomes very dangerous to read, which will not be obvious at all.

Yeah... I get your point... I have a feeling this complexity of managing server/pending values should be abstract in the field itself somehow, but this is easier said than done...

Let's keep `undefined` and have `isPendingPinned` name then, even if that's not a pure boolean.